ostelco-ios-client/ViewControllers/Home/HomeViewController.swift

```diff
@@ -48,12 +48,21 @@ class HomeViewController: ApplePayViewController {
     }()
 
     private func showMessage() {
-        welcomeLabel.isHidden = false
-        messageLabel.isHidden = false
-        DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) { [weak self] in
-            self?.welcomeLabel.isHidden = true
-            self?.messageLabel.isHidden = true
-        }
+        welcomeLabel.alpha = 1.0
+        messageLabel.alpha = 1.0
+        UIView.animate(
+            withDuration: 2.0,
+            delay: 2.0,
+            options: .curveEaseIn,
+            animations: {  [weak self] in
+                self?.hideMessages()
+            },
+            completion: nil)
+    }
+
+    func hideMessages() {
+        welcomeLabel.alpha = 0.0
+        messageLabel.alpha = 0.0
```

Comment:
FWIW I'd just shove these into the `animations` block but if you'd rather leave 'em here, I'm good with that. Probably worth making this method `private` if you do though